pkg/ext-proc/backend/inferencepool_reconciler.go

```diff
@@ -29,26 +29,31 @@ type LLMServerPoolReconciler struct {
 	Zone           string
 }
 
-func (c *LLMServerPoolReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+func (c *InferencePoolReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	if req.NamespacedName.Name != c.ServerPoolName && req.NamespacedName.Namespace != c.Namespace {
 		return ctrl.Result{}, nil
 	}
-	klog.V(2).Infof("Reconciling LLMServerPool %v", req.NamespacedName)
+	klog.V(1).Info("reconciling InferencePool", req.NamespacedName)
 
-	serverPool := &v1alpha1.LLMServerPool{}
+	serverPool := &v1alpha1.InferencePool{}
 	if err := c.Get(ctx, req.NamespacedName, serverPool); err != nil {
-		klog.Errorf("Unable to get LLMServerPool: %v", err)
+		klog.Error(err, "unable to get InferencePool")
 		return ctrl.Result{}, err
 	}
 
-	klog.V(2).Infof("Updated LLMServerPool: %+v", serverPool)
-	c.Datastore.setLLMServerPool(serverPool)
+	c.updateDatastore(serverPool)
```

Comment:
`c.Datastore.setInferencePool(serverpool)`?

Reply:
Called within `updateDatastore`

Reply:
pls use `setInferencePool`, it has the RWMutex. Otherwise may run into race condition